static/src/javascripts/projects/common/modules/navigation/navigation.js

```diff
@@ -11,37 +11,74 @@ define([
     mediator,
     $
 ) {
-    var Navigation = {
-        init: function () {
-            this.addMegaNavMenu();
-            this.enableMegaNavToggle();
-            this.replaceAllSectionsLink();
-        },
-
-        addMegaNavMenu: function () {
-            var megaNav     = $('.js-transfuse'),
-                placeholder = $('.' + megaNav.attr('data-transfuse-target'));
-
-            fastdom.write(function () {
-                placeholder.html(megaNav.html());
-            });
-        },
-
-        replaceAllSectionsLink: function () {
-            $('.js-navigation-header .js-navigation-toggle').attr('href', '#nav-allsections');
-        },
-
-        enableMegaNavToggle: function () {
-            bean.on(document, 'click', '.js-navigation-toggle', function (e) {
-                var target = $('.' + e.currentTarget.getAttribute('data-target-nav'));
-
-                e.preventDefault();
-                fastdom.write(function () {
-                    target.toggleClass('navigation--expanded navigation--collapsed');
-                    mediator.emit(target.hasClass('navigation--expanded') ? 'modules:nav:open' : 'modules:nav:close');
-                });
-            });
-        }
+    function Navigation() {
+        this.isInserted = false;
+    }
+
+    Navigation.prototype.init = function () {
+        this.eatMegaNav();
+        this.enableMegaNavToggle();
+        this.replaceAllSectionsLink();
+        this.$headerNav = $('.js-navigation-header');
+    };
+
+    Navigation.prototype.eatMegaNav = function () {
+        var $megaNav = $('.js-transfuse');
+
+        this.megaNavHtml = $megaNav.html();
+
+        fastdom.write(function () {
+            $megaNav.remove();
+        });
+    };
+
+    Navigation.prototype.replaceAllSectionsLink = function () {
+        $('.js-navigation-toggle').attr('href', '#nav-allsections');
+    };
+
+    Navigation.prototype.scrollToHeaderNav = function () {
+        var that = this;
+
+        fastdom.read(function () {
+            window.scrollTo(0, that.$headerNav.offset().top);
```

Comment:
The nav isn't at 0, 0, though? So it is a UX hit, surely? The ad can be quite large.